pyfr/plugins/tavg.py

```diff
@@ -166,44 +253,98 @@ def __call__(self, intg):
+                    comm.Reduce(max_dev, None, op=mpi.MAX, root=root)
+                    comm.Reduce(acc_dev, None, root=root)
+                else:
+                    comm.Reduce(mpi.IN_PLACE, max_dev, op=mpi.MAX, root=root)
+                    comm.Reduce(mpi.IN_PLACE, acc_dev, root=root)
+                            
+                if self.fexprs: 
+                    
+                    # Evaluate functional expressions and deviations
+                    funex, fdev = self._eval_fun_exprs_var(dev, tavg)
+
+                    # Maximum and sum of functional deviations
+                    max_fdev = np.zeros_like(self.fnames, dtype=np.float64)
+                    acc_fdev = np.zeros_like(self.fnames, dtype=np.float64) 
+                    for fx in fdev:
+                        max_fdev = np.array([max(max_fdev[i], np.amax(f)) 
+                                    for i, f in enumerate(fx.swapaxes(0, 1))])
+                        acc_fdev = np.array([sum(sum(f), acc_fdev[i]) for 
+                                    i, f in enumerate(fx.swapaxes(0, 1))])
+
+                    if rank != root:
+                        comm.Reduce(max_fdev, None, op=mpi.MAX, root=root)
+                        comm.Reduce(acc_fdev, None, root=root)
+                    else:
+                        comm.Reduce(mpi.IN_PLACE, max_fdev, op=mpi.MAX, root=root)
+                        comm.Reduce(mpi.IN_PLACE, acc_fdev, root=root)
+
+                    # Stack the functional expressions
                     tavg = [np.hstack([a, f]) for a, f in zip(tavg, funex)]
-
+   
+                if self.dev_mode == 'all':
+                    # Stack std deviations
+                    tavg = [np.hstack([a, d]) for a, d in zip(tavg, dev)]
+                    if self.fexprs:
+                        # Stack functional deviations
+                        tavg = [np.hstack([a, df]) for a, df in zip(tavg, fdev)]
+     
                 # Form the output records to be written to disk
                 data = dict(self._ele_region_data)
+
                 for (idx, etype, rgn), d in zip(self._ele_regions, tavg):
                     data[etype] = d.astype(self.fpdtype)
 
                 stats = Inifile()
                 stats.set('data', 'prefix', 'tavg')
                 stats.set('data', 'fields', ','.join(self.outfields))
-                stats.set('tavg', 'tstart', tstart)
+                stats.set('tavg', 'tstart', self.tstart_acc)
                 stats.set('tavg', 'tend', intg.tcurr)
+
+                # Write summarised stats   
+                if rank == root:
+                    # Write std deviations
+                    for an, vm, vc in zip(self.anames, max_dev, acc_dev):
+                        stats.set('tavg', f'avg-std-dev-{an}', vc/tpts)
+                        stats.set('tavg', f'max-std-dev-{an}', vm)
+                    
+                    # Followed by functional deviations
+                    if self.fexprs:
```

Comment:
Not needed.  Just run the for loop.  If `fnames` is empty then `zip` will yield nothing.